tidy3d/components/medium.py

```diff
@@ -3092,6 +3184,49 @@ def loss_upper_bound(self) -> float:
         ep = ep[~np.isnan(ep)]
         return max(ep.imag)
 
+    def compute_derivatives(self, derivative_info: DerivativeInfo) -> AutogradFieldMap:
+        """Compute adjoint derivatives for each of the ``fields`` given the multiplied E and D."""
+
+        # compute all derivatives beforehand
+        dJ_deps = self.derivative_eps_complex_volume(
+            E_der_map=derivative_info.E_der_map, bounds=derivative_info.bounds
+        )
+
+        dJ_deps = complex(dJ_deps)
+
+        # TODO: fix for multi-frequency
+        frequency = derivative_info.frequency
+        poles_complex = [(complex(a), complex(c)) for a, c in self.poles]
+        poles_complex = np.stack(poles_complex, axis=0)
```

Comment:
Maybe just `poles_complex = np.array([(complex(a), complex(c)) for a, c in self.poles])`? `np.stack` is identical to `np.array` if `axis=0`.

Reply:
hm, sort of like the explicit axis because it helps me keep track of the fact that the poles are getting stacked along the first axis. but maybe array is fine.